### PR TITLE
[WIP] CI: Add CMake CUDA Build

### PR DESCRIPTION
**Pull request by ax3l**
GPU, GPUuuu :champagne: 
(Build only.)

cc @mic84 @maxpkatz @atmyers

### CODE REVIEW

File: .github/workflows/cmake/dependencies_nvcc.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    gfortran-5          \
+    libopenmpi-dev      \
+    openmpi-bin         \
+    nvidia-cuda-dev     \
+    nvidia-cuda-toolkit
+
+# Patch broken GCC 5.5 libs in <algorithm>
+#   https://gcc.gnu.org/bugzilla/show_bug.cgi?id=76731
+#   https://stackoverflow.com/a/50815334/2719194
+for f in avx512fintrin.h avx512pfintrin.h avx512vlintrin.h
```
Comment (ax3l) [inline .github/workflows/cmake/dependencies_nvcc.sh]: I wish this wasn't necessary, but:
- CUDA 9.1 was incompatible with `std::tuple` in GCC 6
- GCC 5.5 ships broken stdlib headers with the issues linked herein

File: Tutorials/GPU/ParallelReduce/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,80 @@
+# Preamble ####################################################################
+#
+cmake_minimum_required(VERSION 3.14.0)
+project(ParallelReduce LANGUAGES CXX Fortran)
+#                                    ^^^^^^^ required: CMake bug
```
Comment (ax3l) [inline Tutorials/GPU/ParallelReduce/CMakeLists.txt]: fixed in #716

File: .github/workflows/cmake/dependencies_nvcc.sh (hunk 1 above)
Comment (ax3l) [reply .github/workflows/cmake/dependencies_nvcc.sh]: Upstream bug report with Ubuntu filed as: https://bugs.launchpad.net/ubuntu/+source/nvidia-cuda-toolkit/+bug/1865091

File: Tutorials/GPU/ParallelReduce/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+#
+# AMReX
+#   preset required, non-default AMReX options
+#   https://amrex-codes.github.io/amrex/docs_html/BuildingAMReX.html#building-with-cmake
+set(ENABLE_CUDA         ON CACHE INTERNAL "")
+set(ENABLE_TINY_PROFILE ON CACHE INTERNAL "")
+
+if(ENABLE_CUDA)
+    list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/../../../Tools/CMake")
+    include(AMReX_SetupCUDA)
```
Comment (mic84) [inline Tutorials/GPU/ParallelReduce/CMakeLists.txt]: Must add enable_language(CUDA) explicitly before including AMReX_SetupCUDA

Comment (ax3l) [reply Tutorials/GPU/ParallelReduce/CMakeLists.txt]: Ref.: https://github.com/AMReX-Codes/amrex/commit/b14219bd0ce6c72638bcd97840cee2bd2179108b

**Comment by ax3l** [comment]
superseded by #721

**Pull request closed without merging**